packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesDevLauncherController.java

```diff
@@ -0,0 +1,116 @@
+package expo.modules.updates;
+
+import android.content.Context;
+
+import androidx.annotation.Nullable;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+
+import expo.modules.updates.db.DatabaseHolder;
+import expo.modules.updates.db.entity.AssetEntity;
+import expo.modules.updates.db.entity.UpdateEntity;
+import expo.modules.updates.launcher.DatabaseLauncher;
+import expo.modules.updates.launcher.Launcher;
+import expo.modules.updates.loader.RemoteLoader;
+import expo.modules.updates.manifest.Manifest;
+import expo.modules.updates.selectionpolicy.ReaperSelectionPolicyDevelopmentClient;
+import expo.modules.updates.selectionpolicy.SelectionPolicy;
+import expo.modules.updatesinterface.UpdatesInterface;
+
+public class UpdatesDevLauncherController implements UpdatesInterface {
+
+  private JSONObject mRawManifest;
+
+  public UpdatesDevLauncherController(Context context) {
+    UpdatesController.initializeInternal(context);
```

Comment:
I'm worried about initializing a global singleton from within a constructor -- it makes UpdatesDevLauncherController not really an instance. seems preferable to create a new UpdatesController instance and store it in mUpdatesController.

Reply:
That makes sense; currently, though, there are other places in the code that rely on UpdatesController being a singleton (notably, UpdatesModule -- through UpdatesService -- which needs access to UpdatesController in order for the module methods to work in published projects). What do you think about instead making UpdatesDevLauncherController also a global singleton? (https://github.com/expo/expo/pull/13032/commits/9150acf7b2b2b96b489c50772c2e02b40fca4b0f)

Reply:
A singleton UpdatesDevLauncherController is likely fine. The benefits of making everything an instance tend to be better testability since you can pass in mock instances rather than mocking out a global singleton, and the ability to make multiple UpdatesDevLauncherControllers if that were to become useful.

The main thing we want to avoid, though, is having unexpected side effects from the constructor. 9150acf looks alright to me.